router/batchrouter/asyncdestinationmanager/klaviyobulkupload/klaviyobulkupload.go

```diff
@@ -0,0 +1,359 @@
+		"module":   "batch_router",
+		"destType": "KLAVIYO_BULK_UPLOAD",
+	})
+	eventsAbortedStat.Count(len(importingJobIds))
+	return common.AsyncUploadOutput{
+		AbortCount:    len(importingJobIds),
+		DestinationID: destinationID,
+		AbortJobIDs:   importingJobIds,
+		AbortReason:   fmt.Sprintf("%s %v", errorString, err.Error()),
+	}
+}
+
+func (kbu *KlaviyoBulkUploader) ExtractProfile(input Input) (Profile, error) {
+	Message := input.Message
+	Body := Message.Body
+	Json := Body.JSON
+	Data := Json.Data
+	Attributes := Data.Attributes
+	profileObject := Attributes.Profiles.Data[0]
+
+	jobIdentifier := profileObject.Attributes.JobIdentifier
+	jobIdentifierArray := strings.Split(jobIdentifier, ":")
+	jobIdentifierValue, _ := strconv.ParseInt(jobIdentifierArray[1], 10, 64)
+	if kbu.jobIdToIdentifierMap == nil {
+		kbu.jobIdToIdentifierMap = make(map[string]int64)
+	}
+	kbu.jobIdToIdentifierMap[jobIdentifierArray[0]] = jobIdentifierValue
+
+	// delete jobIdentifier from the attributes map as it is not required in the final payload
+	profileObject.Attributes.JobIdentifier = ""
+
+	return profileObject, nil
+}
+
+func (kbu *KlaviyoBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	startTime := time.Now()
+	destination := asyncDestStruct.Destination
+	filePath := asyncDestStruct.FileName
+	importingJobIDs := asyncDestStruct.ImportingJobIDs
+	destType := destination.DestinationDefinition.Name
+	destinationID := destination.ID
+	listId, _ := destination.Config["listId"].(string)
+	statLabels := stats.Tags{
+		"module":   "batch_router",
+		"destType": destType,
+	}
+	file, err := os.Open(filePath)
+	if err != nil {
+		return kbu.generateKlaviyoErrorOutput("Error while opening file. ", err, asyncDestStruct.ImportingJobIDs, destinationID)
+	}
+	defer file.Close()
+	var combinedProfiles []Profile
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var input Input
+		line := scanner.Text()
+		err := json.Unmarshal([]byte(line), &input)
+		if err != nil {
+			return kbu.generateKlaviyoErrorOutput("Error while parsing JSON.", err, importingJobIDs, destinationID)
+		}
+		profileStructure, err := kbu.ExtractProfile(input)
+		if err != nil {
+			return kbu.generateKlaviyoErrorOutput("Error while extracting profiles.", err, importingJobIDs, destinationID)
+		}
+		combinedProfiles = append(combinedProfiles, profileStructure)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return kbu.generateKlaviyoErrorOutput("Error while reading file.", err, importingJobIDs, destinationID)
+	}
+
+	combinedPayload := createFinalPayload(combinedProfiles, listId)
+
+	// Convert combined payload to JSON
+	outputJSON, err := json.MarshalIndent(combinedPayload, "", "  ")
+	if err != nil {
+		return kbu.generateKlaviyoErrorOutput("Error while marshaling combined JSON.", err, importingJobIDs, destinationID)
+	}
+	uploadURL := "https://a.klaviyo.com/api/profile-bulk-import-jobs/"
```

Comment:
Can we create a constant variable for this URL? We are using it in some other places.